features/dlg_crash_info/README.md

```diff
@@ -0,0 +1,253 @@
+3) Call dci_get_data() to retrieve the reset data available.  
+4) Clear the tracked data, to allow for tracking of new resets.  Once the data has reached its intended target, the user should clear the data, to allow for tracking of more resets.  If the buffer is full, no more resets will be tracked. 
+5) Implement the optional epoch callback.  When the module is finished parsing a reset, it will call into the WEAK function dci_get_epoch().  The user can implement a uint32_t value here, to assign to the reset buffer.   
+6) Storing this to NVM is preferred for this type of data and would need to be implemented by the user.   In this case, the user should set DCI_NUM_RESETS (1), after the application is initialized, call dci_get_data() and store this in NVM.   The application should then clear this data immediately with dci_clear_data().  Retrieval and storage is left to the user.  
+
+### *DCI Configuration*
+There is a configuration file dci_config.h that allows for simple configurations
+- DCI_RAMAP_ADDR0_TEST:  This modifies the software reset and boot sequence to boot back into RAM.  Simplifies debugging of the application for continous development of the module. 
+> __Note:__ 
+**WARNING!!!** This is used for testing RAM builds only, for debugging and development of the DCI Module.  This SHOULD be disabled once the application is stored into NVM.  
+- DCI_CALL_DEPTH:  Stack trace buffer length.  Length is equal to DCI_CALL_DEPTH * sizeof(uint32_t).  Default is set to 6 as this empirically gives enough history to solve the majority of issues.  If constrained on memory, the user can lower this value.  
+- DCI_NUM_RESETS:  Should be set to 1 if storing this in Non Volatile memory.  In the case of it being retained in RAM, the user can set the number of resets that can be stored at once.  
+- DCI_PRINTF:  Application uart printing mechanism.  Default set to arch_printf() to enable native debug printing from the SDK.  
+- DCI_PRINT_VERBOSE:  Enables the API for dci_print_verbose.   Calling into dci_print_verbose will deserialize the data and print it in a readable format.  This functionality can take up additionally code and data space.  
+- ENABLE_DCI_SERVICE:  Enables the native custom service for offloading the DCI data.  
+
+## DCI Data types
+
+Each data type used in this application, can be referred to in dci_types.h.
+dci_get_data returns a type of dci_info_t
+
+```c
+typedef __PACKED_STRUCT
+{
+	uint8_t			header;
+	uint16_t 		length;
+	dci_data_t	    *data;		
+	
+}dci_info_t;
+```
+
+- header: Sanity check for data retrieval.  Values is defined by DCI_HEADER.
+- length:  *data length, to allow for proper buffer allocation.  
+- data:  dci_data_t data type
+
+```c
+typedef __PACKED_STRUCT
+{
+	dci_reset_reason_t		    last_reset_reason;
+	uint8_t						num_resets;
+	dci_fault_info_t			fault_data[DCI_NUM_RESETS_STORED];
+	
+}dci_data_t;
```

Comment:
Field name alignment is off when viewing README on Github

Reply:
fixed alignment